packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1184,12 +1184,30 @@ class RenderEditable extends RenderBox {
 
   /// Select a word around the location of the last tap down.
   void selectWord({@required SelectionChangedCause cause}) {
+    selectWordsInRange(from: _lastTapDownPosition, cause: cause);
+  }
+
+  /// Selects the set words of a paragraph in a given range of global positions.
+  ///
+  /// The first and last endpoints of the selection will always be at the
+  /// beginning and end of a word respectively.
+  void selectWordsInRange({@required Offset from, Offset to, @required SelectionChangedCause cause}) {
     assert(cause != null);
     _layoutText(constraints.maxWidth);
     assert(_lastTapDownPosition != null);
     if (onSelectionChanged != null) {
-      final TextPosition position = _textPainter.getPositionForOffset(globalToLocal(_lastTapDownPosition));
-      onSelectionChanged(_selectWordAtOffset(position), this, cause);
+      final TextPosition firstPosition = _textPainter.getPositionForOffset(globalToLocal(from));
```

Comment:
While testing the other PR, I just noticed this needs to be `from/to + -_paintOffset`. Otherwise it gets wonky when it's clipped and scrollable.

Reply:
interesting, ok, I'll change that